### PR TITLE
[ENH] Scatter Plot: Add ellipse/s

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

Implements #6734

##### Description of changes
Add ellipse (similar to regression line).

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by VesnaT** [comment]
https://github.com/biolab/orange3/assets/11465003/44f2d794-10f9-49fa-878c-4deb95f3daec

**Comment by thocevar** [comment]
```
vals, vects = np.linalg.eig(cov)
...
cx = np.cos(m) * np.sqrt(vals[0] * f)
cy = np.sin(m) * np.sqrt(vals[1] * f)
```
Previously, the stretch of the ellipse was `cov[0, 0]` which is the variance along each coordinate axis while the stretch should reflect the variance in the direction of the computed eigenvectors. This results in more appropriate shapes. However, the regression line will not be completely in line with the ellipse, because it's minimizing vertical errors and not perpendicular projections.

**Comment by janezd** [comment]
Lines can also minimize perpendicular distance if you set the plot to treat variables as independent.

**Comment by thocevar** [comment]
By using the option "treat variables as independent" the regression line actually lines up with the ellipse. Do we want to prioritize this?

**Comment by janezd** [comment]
By making it a default? Probably not because users expect a regression line. Yet, scatter plot shows relation between two variables, so perpendicular projection *should* be the default. We never found a good solution to make users aware that there are two possibilities and that they must decide what they want.

File: Orange/widgets/visualize/owscatterplot.py [hunk 1]
```diff
@@ -274,8 +298,41 @@ def update_regression_line(self):
...
+        angle = math.atan2(vects[1, 0], vects[0, 0])
+        matrix = np.array([[np.cos(angle), -np.sin(angle)],
+                           [np.sin(angle), np.cos(angle)]])
+
+        n = len(x)
+        f = ss.f.ppf(0.95, 2, n - 2)
+        f = f * 2 * (n - 1) / (n - 2)
+        m = [np.pi * i / 100 for i in range(201)]
+        cx = np.cos(m) * np.sqrt(cov[0, 0] * f)
+        cy = np.sin(m) * np.sqrt(cov[1, 1] * f)
```
Comment (lanzagar) [inline Orange/widgets/visualize/owscatterplot.py:319]: Use @thocevar 's fix.

**Review by lanzagar — COMMENTED** [review]
After the fix everything looks ok.
As discussed we should maybe move this to the base class.

**Pull request merged**